packages/kit/src/runtime/client/router.js

```diff
@@ -185,7 +185,8 @@ export class Router {
 			if (a instanceof SVGAElement ? a.target.baseVal : a.target) return;
 
 			// Check if new url only differs by hash
```

Comment:
```suggestion
			// Check if new url only differs by hash and use the browser default behavior in that case
			// This will ensure the `hashchange` event is fired
			// Removing the hash does a full page navigation in the browser, so make sure a hash is present
```